cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -542,8 +542,10 @@ class device_row_comparator {
                                                 size_type const rhs_index) const noexcept
   {
     int last_null_depth = std::numeric_limits<int>::max();
-    size_type list_column_index{0};
+    size_type list_column_index{-1};
     for (size_type i = 0; i < _lhs.num_columns(); ++i) {
+      if (_lhs.column(i).type().id() == type_id::LIST) { ++list_column_index; }
```

Comment:
Previously `list_column_index` is not increased at line 550 below. Thus, if early exit is called at line 550, the value of `list_column_index` is not updated properly.